### PR TITLE
Add the IOTICS circle services

### PR DESCRIPTION
**Pull request by StephanieBracaloni**
Add IOTICS circle services:
- upsert
- describe
- delete
- list all

The remaining one will be part of another PR

### CODE REVIEW

File: proto/iotics/api/circle.proto [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+option php_namespace = "Iotics\\Api";
+
+// ---------------------------------------------------------------------------------------------------------------------
+
+
+// CircleAPI enables creation and management of Iotics circles.
+// Services only affect local resources.
+service CircleAPI {
+
+  // UpsertCircle creates or update a circle with its metadata.
```
Comment (vtermanis) [inline proto/iotics/api/circle.proto]: > Creates or update**s**

(same for twin)

File: proto/iotics/api/circle.proto [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+  // The full state is applied (ie. if the operation succeeds the state of the circle will
+  // be the one described in the payload)
+  rpc UpsertCircle(UpsertCircleRequest) returns (UpsertCircleResponse) {}
+
+  // DeleteCircle deletes a circle.
+  rpc DeleteCircle(DeleteCircleRequest) returns (DeleteCircleResponse) {}
+
+  // Describes a circle.
```
Comment (vtermanis) [inline proto/iotics/api/circle.proto:33]: (pedantic) Based on how the other ones are written, should this and the below one say e.g.:
"DescribeCircle describes a circle"?

File: proto/iotics/api/circle.proto [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+  rpc ListAllCircles(ListAllCirclesRequest) returns (ListAllCirclesResponse) {}
+}
+
+message CircleID {
+  // Circle Identifier (using DID format)
+  string id = 1;
+}
+
+
+message UpsertCircleRequest {
```
Comment (vtermanis) [inline proto/iotics/api/circle.proto:46]: (unimportant) some of the other definitions also have a docstring for each message type. (So I added them for e.g. the ACL PR). But I also appreciate that they are self-explanatory.

**Review by vtermanis — APPROVED** [review]
LGTM :heavy_check_mark: 

V. minor docstring observations only.

**Review by captnz — APPROVED** [review]
lgtm

File: proto/iotics/api/circle.proto (hunk 2 above)
Comment (StephanieBracaloni) [reply proto/iotics/api/circle.proto:33]: Before this was mandatory due to the previous linter we were using. But this information is redundant and annoying to maintain.

File: proto/iotics/api/circle.proto (hunk 3 above)
Comment (StephanieBracaloni) [reply proto/iotics/api/circle.proto:46]: Yes same. 
We prefer to go with consistency over "better" approach for the logic but we can stop adding redundant comments

**Comment by StephanieBracaloni** [comment]
Thanks for the reviews

**Pull request merged**